components/Forms/AddServerForm.vue

```diff
@@ -52,14 +54,29 @@ export default Vue.extend({
           required: true,
           mustBeUrl: true
         }
-      }
+      },
+      previousServerLength: 0
     };
   },
+  beforeMount() {
+    this.previousServerLength = this.$store.state.servers.serverList.length;
+  },
   methods: {
     ...mapActions('servers', ['connectServer']),
     async connectToServer(): Promise<void> {
       this.loading = true;
-      await this.connectServer(this.serverUrl);
+      try {
+        await this.connectServer(this.serverUrl);
+        if (this.previousServerLength === 0) {
+          this.$router.push('/login');
+        } else {
+          this.$router.push('/selectserver');
+        }
```

Comment:
```suggestion
        const nbr = this.$store.state.servers.serverList.length;
        await this.connectServer(this.serverUrl);
        if (nbr === 0) {
          this.$router.push('/login');
        } else {
          this.$router.push('/selectserver');
        }
```
I'd also replace the variable in the v-if by `this.$store.state.servers.serverList.length` in order to remove it

Reply:
That would make the button appear, as the transition will not have ended before the store is populated.

Reply:
Fair enough, maybe add a comment there to explicit what's being done  :)